renku/core/util/util.py

```diff
@@ -34,8 +38,63 @@ def to_semantic_version(value: str) -> Optional[Version]:
         return None
 
 
-NoValueType = NewType("NoValueType", object)
-"""Type to represent a value not being set in cases where ``None`` is a valid value."""
+def is_uuid(value):
+    """Check if value is UUID4.
 
-NO_VALUE = NoValueType(object())
-"""Sentinel to represent a value not being set in cases where ``None`` is a valid value."""
+    Copied from https://stackoverflow.com/questions/19989481/
+    """
+    try:
+        uuid_obj = uuid.UUID(value, version=4)
+    except ValueError:
+        return False
+
+    return str(uuid_obj) == value
+
+
+def parallel_execute(function: Callable[..., List[Any]], *, data: List[Any], rate: float = 1, **kwargs) -> List[Any]:
```

Comment:
Should we have a way to un-parallelize this in tests? Does it cause error to be swallowed/more difficult to debug?